test/unit/math/prim/scal/meta/is_eigen_test.cpp

```diff
@@ -0,0 +1,19 @@
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/math/prim/scal/meta/is_eigen.hpp>
+#include <Eigen/Sparse>
+#include <gtest/gtest.h>
+#include <vector>
+
+TEST(MathMeta, primitive) {
+  using stan::is_eigen;
+  EXPECT_FALSE((is_eigen<bool>::value));
+  EXPECT_FALSE((is_eigen<double>::value));
+  EXPECT_FALSE((is_eigen<int>::value));
+
+  EXPECT_FALSE((is_eigen<std::vector<double>>::value));
```

Comment:
Ummm, so it would be good to check that at this poinat in scal is_eigen reports false even on Eigen types. But if that's how the rest of the tests are I can remove this.

`is_vector` does not check the alt condition of a vector coming out as false in scal so maybe pulling in Eigen and vector here is bad